webhooks/framework.go

```diff
@@ -0,0 +1,55 @@
+package webhooks
+
+import (
+	"fmt"
+	jwt "github.com/dgrijalva/jwt-go"
+	"github.com/rancher/webhook-service/drivers"
+)
+
+// ConstructPayload constructs jwt url
+func ConstructPayload(input map[string]interface{}) (map[string]interface{}, error) {
+	if _, ok := input["driver"].(string); !ok {
+		return nil, fmt.Errorf("Driver of type string not provided\n")
+	}
+	driverId := input["driver"].(string)
+	driver := drivers.GetDriver(driverId)
+	if driver == nil {
+		return nil, fmt.Errorf("Driver %s is not registered\n", driverId)
+	}
+	payload, err := driver.ConstructPayload(input)
```

Comment:
Oh, are you referring to the discussion on why the ServiceScaler driver needs a ConstructPayload function when it just creates jwt token? I added it because for the drivers we add later we might have to do some other processing like we discussed. I just thought it'll be uniform. I forgot to discuss this sorry.